src/api/tests/test_available.py

```diff
@@ -0,0 +1,80 @@
+"""Test the available domain API."""
+
+import json
+
+from django.contrib.auth import get_user_model
+from django.test import TestCase, RequestFactory
+
+from ..views import available, _domains, in_domains
+
+
+class AvailableViewTest(TestCase):
+
+    """Test that the view function works as expected."""
+
+    def setUp(self):
+        self.user = get_user_model().objects.create(username="username")
+        self.factory = RequestFactory()
+
+    def test_view_function(self):
+        request = self.factory.get("/available/test.gov")
+        request.user = self.user
+        response = available(request, domain="test.gov")
+        # has the right text in it
+        self.assertContains(response, "available")
+        # can be parsed as JSON
+        response_object = json.loads(response.content)
+        self.assertIn("available", response_object)
+
+    def test_domain_list(self):
+        """Test the domain list that is returned."""
+        domains = _domains()
+        self.assertIn("gsa.gov", domains)
+        # entries are all lowercase so GSA.GOV is not in the set
+        self.assertNotIn("GSA.GOV", domains)
+        self.assertNotIn("igorville.gov", domains)
+        # all the entries have dots
+        self.assertNotIn("gsa", domains)
```

Comment:
Since we are expecting this implementation to change, I wouldn't waste any time on mocking the GH csv, just noting for future external integrations that we'll want to mock external dependencies in the tests.